sources/engine/Xenko.Voxels/Voxels/GraphicsCompositor/VoxelRenderFeature.cs

```diff
@@ -0,0 +1,182 @@
+
+        [DataMember]
+        public RenderStage VoxelizerRenderStage { get; set; }
+        
+        private LogicalGroupReference VoxelizerStorerCasterKey;
+        private StaticObjectPropertyKey<RenderEffect> renderEffectKey;
+
+        protected override void InitializeCore()
+        {
+            base.InitializeCore();
+            renderEffectKey = ((RootEffectRenderFeature)RootRenderFeature).RenderEffectKey;
+            VoxelizerStorerCasterKey = ((RootEffectRenderFeature)RootRenderFeature).CreateViewLogicalGroup("VoxelizerStorer");
+        }
+
+        public override void PrepareEffectPermutations(RenderDrawContext context)
+        {
+            renderVoxelVolumeData = Context.VisibilityGroup.Tags.Get(CurrentProcessedVoxelVolumes);
+            if (renderVoxelVolumeData == null) return;
+
+
+            var renderEffects = RootRenderFeature.RenderData.GetData(renderEffectKey);
+            int effectSlotCount = ((RootEffectRenderFeature)RootRenderFeature).EffectPermutationSlotCount;
+
+
+            var rootEffectRenderFeature = ((RootEffectRenderFeature)RootRenderFeature);
+
+            if (rootEffectRenderFeature == null) return;
+
+            foreach (var processedVolumeKeyValue in renderVoxelVolumeData)
+            {
+                var processedVolume = processedVolumeKeyValue.Value;
+
+                if (processedVolume == null) continue;
+
+                foreach (VoxelizationPass pass in processedVolume.passList.passes)
+                {
+                    var viewFeature = pass.view.Features[RootRenderFeature.Index];
+                    if (processedVolume == null)
+                        continue;
+
+                    pass.storer.UpdateVoxelizationLayout("Storage");
+                    for (int i  = 0; i < pass.AttributesIndirect.Count;  i++)
+                    {
+                        var attr = pass.AttributesIndirect[i];
+                        attr.UpdateVoxelizationLayout("AttributesIndirect["+i+"]");
+                    }
+
+                    var effectSlot = rootEffectRenderFeature.GetEffectPermutationSlot(RenderSystem.RenderStages[pass.view.RenderStages[0].Index]);
+
+                    foreach (var renderObject in pass.view.RenderObjects)
+                    {
+                        var staticObjectNode = renderObject.StaticObjectNode;
+                        if (staticObjectNode == null)
+                            continue;
+
+                        var staticEffectObjectNode = staticObjectNode * effectSlotCount + effectSlot.Index;
+                        if (staticEffectObjectNode == null)
+                            continue;
+
+                        RenderEffect renderEffect = null;
+                        try
+                        {
+                            renderEffect = renderEffects[staticEffectObjectNode];
+                        }
+                        catch
+                        {
```

Comment:
Yeah it would throw with an OutOfRange whenever the Game Studio started up. I haven't been able to debug the Game Studio from VS, so I put this in as a workaround. I'll see if I can find the problem properly.